internal/provider/kubernetes/predicates.go

```diff
@@ -168,7 +168,20 @@ func (r *gatewayAPIReconciler) validateServiceForReconcile(obj client.Object) bo
 		r.log.Info("unexpected object type, bypassing reconciliation", "object", obj)
 		return false
 	}
+	labels := svc.GetLabels()
+	gclass, ok := labels[gatewayapi.OwningGatewayClassLabel]
+	if ok {
+		gtw := r.findGateways(ctx, gclass)
```

Comment:
should we instead check if this gclass is the accepted gclass in the providerResources watchable message ?
and if so  just fetch the gateways from the message ?

Reply:
yes, that is much simpler, thanks :)